cmd/promoted-image-governor/main.go

```diff
@@ -399,16 +479,51 @@ func main() {
 		return
 	}
 
-	kubeConfig, err := clientcmd.BuildConfigFromFlags("", opts.kubeconfig)
+	kubeconfigs, err := opts.kubernetesOptions.LoadClusterConfigs()
 	if err != nil {
-		logrus.WithError(err).Fatalf("could not load kube config from path %s", opts.kubeconfig)
+		logrus.WithError(err).Fatal("failed to load kubeconfigs")
+	}
+
+	inClusterConfig, hasInClusterConfig := kubeconfigs[kube.InClusterContext]
+	delete(kubeconfigs, kube.InClusterContext)
+	delete(kubeconfigs, kube.DefaultClusterAlias)
+
+	if _, hasAppCi := kubeconfigs[appCIContextName]; !hasAppCi {
+		if !hasInClusterConfig {
+			logrus.WithError(err).Fatalf("had no context for '%s' and loading InClusterConfig failed", appCIContextName)
+		} else {
+			// There is a corner case that the inClusterConfig is the current context of a loaded kubeconfig file
```

Comment:
Change dev. We should never, ever write code in production to handle dev cases. Make dev look like prod.

Reply:
Simplified it a bit.
The `else-branch` here is to ensure if the inClusterConfig is correct because it could be the one for the current context of the last loaded kubeconfig file.

If we do not consider the Dev env at all, the code would be

```
         inClusterConfig := kubeconfigs[kube.InClusterContext]
	delete(kubeconfigs, kube.InClusterContext)
	delete(kubeconfigs, kube.DefaultClusterAlias)
	kubeconfigs[appCIContextName] = inClusterConfig
```

However, I cannot figure out a way to make the above code work for dev because dev does not run in pod, and have no inclusterconfig.

Reply:
Why do we need it? In dev, pass a `$KUBECONFIG` with the `app.ci` context explicitly and do not use in-cluster.